### ISSUE TITLE
Challenge metrics

### ISSUE THREAD

**Issue by lzdoesdata** [lforando/mini-project-3#5]
Now that we have plotted our desired shape files (landmarks, trails, boundary, contour), we need to devise metrics for determining what attributes make something "difficult". Here's some ideas:
1. how quickly the elevation changes over a certain length of ft
2.length of trail
3.age friendly?
etc.

let me know what you guys think!

**Comment by lzdoesdata** [comment]
the process for this includes:
#need to create 3 variables of difficulty
#(adding difficulty column to macleish layers so you can map aesthetic)

**Comment by lzdoesdata** [comment]
some official guidelines from nps.org:
"Our trails difficulty ratings are generally based on trail condition, steepness of grades, gain and loss of elevation, and the amount and kinds of natural barriers that must be traversed.
These ratings mean different things depending on if the trail was designed for foot use, for horse use, for bike use or for wheelchair use."

**Comment by sarah-daum** [comment]
So the code for computing length is similar to what we completed in lab 16: 

`library(sf)
macleish_layers %>%
  pluck("trails") %>%
  st_length()`

Then create a new object: 

`trails <- macleish_layers %>%
  pluck("trails") %>%
  mutate(computed_length = st_length(geometry))`

This new data frame creates a minor problem: the names of the trails here are different than the ones on the map provided by MacLeish. Some of the trails are repeated twice, with different lengths. I don't know why this is, but I'm thinking we should estimate which trail in "trails" corresponds with the same trail on the map.

**Comment by sarah-daum** [comment]
Also @lzdoesdata you talked about converting meters to miles: to do that we have to get rid of the [m] that appears next to the values in the computed_length column. Here's the code for that: 

`trails$computed_length <-  as.numeric(str_extract(trails$computed_length, "[0-9]+"))`

**Comment by sarah-daum** [comment]
I'm unclear how we are going to actually rate the difficulty of each trail. There are several different formulas to measure it [here](https://outdoors.stackexchange.com/questions/20430/is-there-a-standard-hiking-trail-difficulty-rating-system-in-the-united-states). I think the [Shenandoah](https://www.nps.gov/shen/planyourvisit/how-to-determine-hiking-difficulty.htm) system might be the most straightforward. 

To summarize their method:

Equation: Elevation Gain x 2 x distance (in miles). The product's square root is the numerical rating. 

**Easiest**
Numerical Rating: less than 50

**Moderate**
Numerical Rating: 50-100

**Moderately Strenuous**
Numerical Rating: 100 –150 

**Strenuous**
Numerical Rating: 150-200 


I doubt we will see any trails that are more than "Moderately Strenuous".

**Comment by lforando** [comment]
trails$computed_length <- as.numeric(str_extract(trails$computed_length, "[0-9]+")) this code is wrong dot use it !
